File: web/src/services/authService.js
import axios from 'axios';

const instance = axios.create({
    baseURL: 'http://localhost:8080/auth'
});

const authService = {
    login: (username, password) => {
        return instance.post('/login', { username, password })
            .then(response => {
                return response.data;
            });
    },
    register: (username, password) => {
        return instance.post('/register', {
            username, password
        });
    },
    logout: () => {
        return instance.post('/logout');
    },
    getCurrentUser: () => {
        return instance.get('/user');
    }
};

export default authService;

File: TFGApp/src/main/java/com/example/tfgapp/Auth/AuthService.java
package com.example.tfgapp.Auth;

import com.example.tfgapp.Entity.User;
import com.example.tfgapp.Jwt.JwtService;
import com.example.tfgapp.Repository.UserRepository;
import org.springframework.security.authentication.AuthenticationManager;
import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
import org.springframework.security.core.userdetails.UserDetails;
import org.springframework.security.crypto.password.PasswordEncoder;
import org.springframework.stereotype.Service;
import lombok.RequiredArgsConstructor;

@Service
@RequiredArgsConstructor
public class AuthService {

    private final UserRepository userRepository;
    private final JwtService jwtService;
    private final PasswordEncoder passwordEncoder;
    private final AuthenticationManager authenticationManager;

    public AuthResponse login(LoginRequest request) {
        authenticationManager.authenticate(new UsernamePasswordAuthenticationToken(request.getUsername(), request.getPassword()));
        UserDetails user=userRepository.findByUsername(request.getUsername()).orElseThrow();
        String token=jwtService.getToken(user);
        return AuthResponse.builder()
                .token(token)
                .username(request.getUsername())
                .build();

    }

    public AuthResponse register(RegisterRequest request) {
        User user = User.builder()
                .username(request.getUsername())
                .password(passwordEncoder.encode( request.getPassword()))
                .build();

        userRepository.save(user);

        return AuthResponse.builder()
                .token(jwtService.getToken(user))
                .username(request.getUsername())
                .build();

    }

}
